### DISCUSSION TITLE
Help with Template Partials

### DISCUSSION THREAD

**Discussion by the-solipsist** [quarto-dev/quarto-cli#6756] [Q&A] [answered]
I'm trying to create an archive of all my writings.  In the title-metadata block, I'd like to add a few fields, i.e., "Publication", "Original URL", "Archived URL", and I'd like to do so with the tabular listings as well.

My understanding is that to do so I need to use [template-partials](https://quarto.org/docs/journals/templates.html#template-partials).

I've sought to do so using these files:
https://gist.github.com/the-solipsist/83b30afcbd2ef951482c91fce95853db

When I run `quarto render 2011-01-25_title.qmd` I get the error:
`Could not find data file templates/_title-meta-author.html`

I've tried placing an exact copy of the `_title-meta-author.html` file in the project base directory, in the `posts/` directory and in a `templates/` directory, but to no avail.  The error goes away if I remove the two instances in `title-metadata.html` in which reference `$_title-meta-author.html()$` (line 8 and line 37), but that, of course, removes the author name from the title-metadata block.

I'm not sure I understand yet how exactly to use template-partials.

**Comment by dragonstyle** [comment, marked as answer]
You're very close- the issue you're running into is that the template/partial system only permits you to provide a specific set of templates (not arbitrary templates). In the case of HTML, you are providing the `title-block` partial which is totally correct, but that partial points to the `_title-meta-author.html` which we don't 'know' about. If you just take the contents of that file and copy and paste it directly into the `title-block.html` file in place of any calls to `_title-meta-author.html()` it should just work.

HTH!

**Reply by FelixBenning** [reply]
Why is it not possible to provide arbitrary templates? I am currently trying to find a way to have a `macros.tex` file where I list `\newcommand{}{}` macros. The issue is, that there is no clean way to include this file in every document.

- https://stackoverflow.com/questions/76398554/how-to-define-latex-macros-globally-for-a-quarto-book
- https://github.com/quarto-dev/quarto-cli/issues/7518#issuecomment-1807808828
- https://github.com/quarto-dev/quarto-cli/discussions/2845

It looks like at the moment you would have to include the `macros.qmd` file manually in every `.qmd` file of the project. My approach was to define a custom `before-title.tex` template where I attempted:

```tex
%before-title.tex
${macros.tex()}
```

I hoped this would then include the `macros.tex` file. Unfortunately this does not work due to this restriction.

**Reply by cderv** [reply]
Hi @FelixBenning, it seems this would be best to open a new discussion about what you are trying to do as the tool as evolved since 2023. 

Let me though try to answer on your question.

> Why is it not possible to provide arbitrary templates?

As explained, Quarto currently needs to know in advance the names of the partials, so only using the defined set of partials can be used. 

So you would need to inline the `macro.tex` inside the `before-title.tex`. LaTeX has a mechanism to include files, though, so you could also use `\input{macro.tex}` in your template. This would work only if `macro.tex` is a standard TeX file, and not a Pandoc template itself. 

> I am currently trying to find a way to have a macros.tex file where I list \newcommand{}{} macros.

If you new commands macros are not using any templating feature, you could use the above `\input` command in a template. But you could also just pass the file as an include. See
https://quarto.org/docs/output-formats/pdf-basics.html#latex-includes

This is useful specifically for defining new commands. 

I hope it helps. Please open a new discussion if you have more to share about your use case.

**Reply by FelixBenning** [reply]
I opened a new discussion: https://github.com/quarto-dev/quarto-cli/discussions/12838. I realized that template partials will not help me with this as they are not fed through pandoc's latex_macros extension.
